### PR TITLE
Seperate version env cache files

### PR DESCRIPTION
**Pull request by pwmckenna**
| Q                        | A 
| ------------------------ | ---
| Patch: Bug Fix?          |  ?
| Major: Breaking Change?  |  ?
| Minor: New Feature?      |  ?
| Deprecations?            |  N
| Spec Compliancy?         |  N
| Tests Added/Pass?        |  N
| License                  | MIT

Because cache entries are keyed on babel version and environment, there's no point in storing different combinations in the same file, because you'll never use the portions of the cache generated with different combinations. For instance, if you use BABEL_ENV values 'node' and 'test', you'll have to load a cache that's twice as big as you'll possibly use.

### CODE REVIEW

File: packages/babel-register/src/cache.js [hunk 1]
```diff
@@ -2,8 +2,11 @@ import path from "path";
 import fs from "fs";
 import { sync as mkdirpSync } from "mkdirp";
 import homeOrTmp from "home-or-tmp";
+import * as babel from "babel-core";
 
-const FILENAME: string = process.env.BABEL_CACHE_PATH || path.join(homeOrTmp, ".babel.json");
+const env = process.env.BABEL_ENV || process.env.NODE_ENV;
```
Comment (xtuc) [inline packages/babel-register/src/cache.js]: You could set the default value here, like:
```js
const env = process.env.BABEL_ENV || process.env.NODE_ENV || "";
```

Comment (pwmckenna) [inline packages/babel-register/src/cache.js]: True, but I don't think that would save us the check below. If neither is set we could end up with something like .babel.6.x.x..json with the extra `.`...which might not be a problem?

Comment (xtuc) [inline packages/babel-register/src/cache.js]: Yes that right. The default environment is `development`:

```js
const env = process.env.BABEL_ENV || process.env.NODE_ENV || "development";
```

Comment (xtuc) [inline packages/babel-register/src/cache.js]: I made #5448 for that purpose

Comment (pwmckenna) [inline packages/babel-register/src/cache.js]: That looks great! I'll keep an eye out and update this when it gets merged in. Thanks for that

Comment (xtuc) [inline packages/babel-register/src/cache.js]: The PR has been merged

Comment (pwmckenna) [inline packages/babel-register/src/cache.js]: Updated!

**Pull request merged**